src/vcu/src/controls.c

```diff
@@ -105,15 +119,70 @@ void execute_controls(void) {
     int32_t limited_torque;
     if (dash_limited_torque < min_sensor_torque) {
       limited_torque = dash_limited_torque;
-    }
-    else {
+    } else {
       limited_torque = min_sensor_torque;
     }
 
     torque_command = limited_torque;
+    bool torque_limited = limited_torque < controls_monitoring.raw_torque;
+
+    if (launch_ctrl_entered = (lc_settings.using_launch_ctrl && lc_state != DONE && !torque_limited)) {
+      // We shouldn't be braking in launch control, so reset regen valve
+      set_brake_valve(false);
+      uint32_t front_wheel_speed = get_front_wheel_speed();
+
+      // Mini FSM
+      switch (lc_state) {
+        case BEFORE:
+          sendSpeedCmdMsg(0, 0);
+
+          // Transition
+          if (pedalbox_min(accel) > LC_ACCEL_BEGIN) {
+            lc_state = SPEEDING_UP;
+            printf("[LAUNCH CONTROL] SPEEDING UP STATE ENTERED\r\n");
+          }
+          break;
+        case SPEEDING_UP:
+          if (torque_command < 1000) sendSpeedCmdMsg(500, torque_command);
```

Comment:
In confluence, we have "LC_SPEEDING_UP_TORQUE", but here it is not a constant.

Reply:
Also can we make 300 and 500 some sort of constants too and add them to the documentation in confluence?

Reply:
Also _technically_ the if statement here with < 1000 should probably be reflected in the state machine somehow. I _think_ you intended to back off on the torque command if we start to exceed 1000, right?

Reply:
Oh yeah I realized I had made it a constant and forgot to push it and the others

Reply:
And yeah my intention we to command max(commanded torque, 1000)